webgpu.h

```diff
@@ -765,6 +764,11 @@ typedef struct WGPURequestAdapterOptions {
     WGPUPowerPreference powerPreference;
 } WGPURequestAdapterOptions;
 
+typedef struct WGPURequiredLimits {
+    WGPUChainedStruct const * nextInChain;
+    WGPUBaseLimits limits;
```

Comment:
This is slightly verbose but I don't see a better way.

WGPULimitsV1 maybe? I don't know what will be the best name in the long run.

Reply:
no hard feelings about the name, "LimitsV1" is fine

Reply:
or just "WGPULimits" as it was originally? and then in the future we can have WGPULimits2, WGPURequiredLimits2, WGPUSupportedLimits2. This is like how Vulkan has VkPhysicalDeviceFeatures and VkPhysicalDeviceFeatures2

Reply:
ok, fiiiiine :)

Reply:
I think the extended limits struct will probably be one per optional feature, rather than having a limits2/etc? May use limits2 for limits we add later (hopefully none, except for webgpu-compat which would also have its own struct).

And if those structs are small then it would be kind of annoying to define three structs for every single one. Wish we had a way to avoid it. We could have supported-limits allocated by the implementation (with a separate `free` call), instead of having a non-const pointer chain? Tradeoff is apps would have to deal with reinterpret_casting by SType and skipping STypes they don't understand.

Reply:
There is a struct-less proposal in https://github.com/webgpu-native/webgpu-headers/pull/102. It sort of introduces a new way to handle optional features catered specifically to limits, but it does help resolve this multi-struct annoyance.